repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -340,14 +351,13 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
 
       def getMatchingUriTenant(doTenantCheck: Boolean, validToken: ValidToken): Option[String] = {
         if (doTenantCheck) {
-          tenantFromUri flatMap { uriTenant =>
-            val tokenTenants = validToken.defaultTenantId.toSet ++ validToken.tenantIds
-            val prefixes = Option(config.getTenantHandling.getValidateTenant.getStripTokenTenantPrefixes).map(_.split('/')).getOrElse(Array.empty[String])
-            tokenTenants find { tokenTenant =>
-              tokenTenant.equals(uriTenant) || prefixes.exists(prefix =>
-                tokenTenant.startsWith(prefix) && tokenTenant.substring(prefix.length).equals(uriTenant)
-              )
-            }
+          val uriTenant = getTenantFromUri
+          val tokenTenants = validToken.defaultTenantId.toSet ++ validToken.tenantIds
+          val prefixes = Option(config.getTenantHandling.getValidateTenant.getStripTokenTenantPrefixes).map(_.split('/')).getOrElse(Array.empty[String])
+          tokenTenants find { tokenTenant =>
+            tokenTenant.equals(uriTenant) || prefixes.exists(prefix =>
+              tokenTenant.startsWith(prefix) && tokenTenant.substring(prefix.length).equals(uriTenant)
+            )
```

Comment:
I think this change goes away if you end up going with my suggestion and letting the lazy val do its thing.